File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework2/Codes/gultekin_omeroktay.rb
# Unconditional Exits in Ruby
# Unlabeled Exits in Ruby
puts("----------------Unlabeled Exits----------------")
for i in 1..6
    if i == 3
      
        puts("Unlabeled continue (next) is used to pass the current iteration when the value is %d" %[i])
        next
    elsif i == 5 
        puts("Unlabeled break is used to break the loop when the value is: %d" %[i])
        puts("The remaining value 6 will not be printed.")
        break
    end
    puts("Loop value: %d" %[i])
end
puts("---------------------------------------------------")

# Labeled Exits in Ruby
# Ruby does not have Labeled Exists but have a unique way of simulating them using catch statement.
# Label is defined with ":" at the catch statement that will be a returning point when the corresponding label is thrown.
    puts("-------------------Labeled Exits-------------------")
    catch :outerBreak do
    for i in 1..3
      catch :innerBreak do
      puts("Outer value: #{i}")
      for i2 in 1..6
        catch :innerContinue do
        if i == 1
          puts(" Labeled continue is used, skipped outer loop iteration value: #{i}")
          throw :innerBreak # inner break = outer continue
        elsif i == 2 and i2 == 3
          puts(" Labeled break is used to terminate current inner loop iteration.")
          throw :innerBreak
        elsif i2 == 3
          puts(" Labeled continue is used, skipped inner loop iteration value: #{i2}")
          throw :innerContinue
        elsif i2 == 5
          puts(" Labeled break is used to terminate current outer loop iteration")
          throw :outerBreak
        end
        puts(" Inner value: #{i2}")
        end
      end
      end
    end
    end
    puts("Outer Loop is successfully broken.")
    puts("---------------------------------------------------")

File: Year 3/Fall Semester/CS-315_Programming_Languages/Homework2/Codes/gultekin_omeroktay.dart
void main() {
  // Unconditional Exits in Dart
  // Unlabeled Exits in Dart
  print("----------------Unlabeled Exits----------------");
  for(var i = 1; i <= 6; i++) {
    if (i == 3) {
      print("Unlabeled continue is used to pass the current iteration when the value is $i");
      continue;
    } else if (i == 5) {
      print("Unlabeled break is used to break the loop when the value is: $i");
      print("The remaining value 6 will not be printed.");
      break;
      }
      print("Loop value: $i");
    }
    print("---------------------------------------------------");

    // Labeled Exits in Dart
    print("-------------------Labeled Exits-------------------");
    Outer: for(var i = 1; i <= 3; i++) {
      print("Outer value: $i");
      Inner: for(var i2 = 1; i2 <= 6; i2++) {
            if(i == 1) {
              print(" Labeled continue is used, skipped outer loop iteration value: $i");
              continue Outer;
            } else if(i == 2 && i2 == 3) {
              print(" Labeled break is used to terminate current inner loop iteration.");
              break Inner;
            } else if(i2 == 3) {
              print(" Labeled continue is used, skipped inner loop iteration value: $i2");
              continue Inner;
            } else if(i2 == 5) {
              print(" Labeled break is used to terminate current outer loop iteration");
              break Outer;
            }
      print(" Inner value: $i2");
      }
    }
    print("Outer Loop is successfully broken.");
    print("---------------------------------------------------");
}